Sources/Classes/Views/BaseCell/BaseCell.swift

```diff
@@ -0,0 +1,170 @@
+    private var topViewRightConstraint: NSLayoutConstraint!
+    private var topViewTopConstraint: NSLayoutConstraint!
+    private var topViewHeightConstraint: NSLayoutConstraint!
+
+    // bottom separator constraints
+    private var bottomViewLeftConstraint: NSLayoutConstraint!
+    private var bottomViewRightConstraint: NSLayoutConstraint!
+    private var bottomViewBottomConstraint: NSLayoutConstraint!
+    private var bottomViewHeightConstraint: NSLayoutConstraint!
+
+    private var topSeparatorInsets    = UIEdgeInsets.zero
+    private var bottomSeparatorInsets = UIEdgeInsets.zero
+
+    private var topSeparatorHeight    = CGFloat(pixels: 1)
+    private var bottomSeparatorHeight = CGFloat(pixels: 1)
+
+    // MARK: - Initialization
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        configureLineViews()
+    }
+
+    public override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        configureLineViews()
+    }
+
+    override open func updateConstraints() {
+        topViewTopConstraint.constant       = topSeparatorInsets.top
+        topViewLeftConstraint.constant      = topSeparatorInsets.left
+        topViewRightConstraint.constant     = topSeparatorInsets.right
+        topViewHeightConstraint.constant    = topSeparatorHeight
+
+        bottomViewLeftConstraint.constant   = bottomSeparatorInsets.left
+        bottomViewRightConstraint.constant  = bottomSeparatorInsets.right
+        bottomViewBottomConstraint.constant = bottomSeparatorInsets.bottom
+        bottomViewHeightConstraint.constant = bottomSeparatorHeight
+
+        super.updateConstraints()
+    }
+
+    private func configureInterface(with viewModel: BaseCellViewModel?) {
+        guard let viewModel = viewModel else {
+            return
+        }
+
+        topView.isHidden           = viewModel.separatorType.topIsHidden
+        bottomView.isHidden        = viewModel.separatorType.bottomIsHidden
+
+        topView.backgroundColor    = viewModel.topSeparatorConfiguration?.color
+        topSeparatorHeight         = viewModel.topSeparatorConfiguration?.height ?? CGFloat(pixels: 1)
+        topSeparatorInsets         = viewModel.topSeparatorConfiguration?.insets ?? .zero
+
+        bottomView.backgroundColor = viewModel.bottomSeparatorConfiguration?.color
+        bottomSeparatorHeight      = viewModel.bottomSeparatorConfiguration?.height ?? CGFloat(pixels: 1)
+        bottomSeparatorInsets      = viewModel.bottomSeparatorConfiguration?.insets ?? .zero
+    }
+
+    private func configureLineViews() {
+        let requiredValues: [Any?] = [
+            topView, bottomView,
+            topViewLeftConstraint, topViewRightConstraint, topViewTopConstraint,
+            bottomViewLeftConstraint, bottomViewRightConstraint, bottomViewBottomConstraint]
+
+        if !requiredValues.contains(where: { $0 == nil }) {
```

Comment:
since you only can call configureLineViews from inits, this check is not required